chain-db-watcher/src/graphql_helpers.ts

```diff
@@ -0,0 +1,127 @@
+import dotenv from 'dotenv';
+import { GraphQLClient } from 'graphql-request';
+
+import { addPostAndProposalMutation, getProposalQuery, loginMutation } from './queries';
+
+dotenv.config();
+
+const graphqlServerUrl = process.env.REACT_APP_HASURA_GRAPHQL_URL;
+
+/**
+ * Tell if there is already a proposal in the DB matching the
+ * onchain proposal id passed as argument
+ *
+ * @param onchain_proposal_id the proposal id that is on chain (not the Prisma db id)
+ */
+export const proposalAlreadyExists = async (onchain_proposal_id: number): Promise<boolean> => {
+
+	if (!graphqlServerUrl) throw new Error ('Environment variable for the REACT_APP_HASURA_GRAPHQL_URL not set');
+
+	try {
+		const client = new GraphQLClient(graphqlServerUrl, {
+			headers: {}
+		});
+
+		return client.request(getProposalQuery, { onchain_proposal_id })
+			.then(data => !!data?.onchain_proposals?.length)
+			.catch(err => {
+				err.response?.errors && console.log('GraphQL response errors',err.response.errors);
+				err.response?.data && console.log('Response data if available',err.response.data);
+				throw new Error(err);
+			});
+
+	} catch (e){
+		console.error('Graphql execution error - Proposal already exists');
+		throw new Error(e);
+	}
+};
+
+/**
+ * Creates a generic post and the linked proposal in hasura discussion DB
+ *
+ * @param proposer address of the proposer of the proposal, encoded with the network prefix
+ * @param onchain_proposal_id the proposal id that is on chain (not the Prisma db id)
+ */
+
+export const addPostAndProposal = async ({ proposer, onchain_proposal_id }: {proposer: string, onchain_proposal_id: number}) => {
+	const DEMOCRACY_TOPIC_ID = 1;
```

Comment:
This is hardcoded. Shouldn't it come from env var?

Reply:
Indeed, since it's something external, I've put it in an env.